api/src/mutations/drop.rs

```diff
@@ -146,17 +95,60 @@ impl Mutation {
         };
 
         let drop_model = drop.insert(conn).await?;
+        let event_key = NftEventKey {
+            id: drop_model.id.to_string(),
+            user_id: user_id.to_string(),
+            project_id: input.project.to_string(),
+        };
 
-        emit_drop_transaction_event(
-            producer,
-            drop_model.id,
-            user_id,
-            serialized_message,
-            signed_message_signatures,
-            input.project,
-            input.blockchain,
-        )
-        .await?;
+        match input.blockchain {
+            BlockchainEnum::Solana => {
+                solana
+                    .event()
+                    .create_drop(event_key, proto::MetaplexMasterEditionTransaction {
+                        collection_id: collection.id.to_string(),
+                        master_edition: Some(proto::MasterEdition {
+                            owner_address,
+                            supply: input.supply.map(TryInto::try_into).transpose()?,
+                            name: metadata_json.name,
+                            symbol: metadata_json.symbol,
+                            metadata_uri: metadata_json.uri,
+                            seller_fee_basis_points: seller_fee_basis_points.into(),
+                            creators: input
+                                .creators
+                                .into_iter()
+                                .map(TryFrom::try_from)
+                                .collect::<Result<_>>()?,
+                        }),
+                    })
+                    .await?;
+            },
+            BlockchainEnum::Polygon => {
+                let amount = input.supply.ok_or(Error::new("supply is required"))?;
+                polygon
+                    .create_drop(event_key, proto::CreateEditionTransaction {
+                        amount: amount.try_into()?,
+                        edition_info: Some(proto::EditionInfo {
+                            // creator should be input.creator
+                            creator: owner_address.clone(),
```

Comment:
@imabdulbasit note to still switch this over. Also should have a validation on creators based on blockchain where solana can have at most 5 and polygon 1.